Also, I am seeing test failures in CI for the test `TestTerragruntInitOnce`:

```
2026-05-08T16:08:27.3322540Z         	Error Trace:	/Users/runner/work/terragrunt/terragrunt/test/integration_test.go:1043
2026-05-08T16:08:27.3327530Z         	Error:      	"\n\x1b[0m\x1b[1mInitializing the backend...\x1b[0m\n\x1b[0m\x1b[1mInitializing modules...\x1b[0m\n\n\x1b[0m\x1b[1mInitializing provider plugins...\x1b[0m\n- Reusing previous version of hashicorp/random from the dependency lock file\n- Using previously-installed hashicorp/random v3.8.0\n\n\x1b[0m\x1b[1m\x1b[32mOpenTofu has been successfully initialized!\x1b[0m\x1b[32m\x1b[0m\n\x1b[0m\x1b[32m\nYou may now begin working with OpenTofu. Try running \"tofu plan\" to see\nany changes that are required for your infrastructure. All OpenTofu commands\nshould now work.\n\nIf you ever set or change modules or backend configuration for OpenTofu,\nrerun this command to reinitialize your working directory. If you forget, other\ncommands will detect it and remind you to do so if necessary.\x1b[0m\n\nOpenTofu used the selected providers to generate the following execution\nplan. Resource actions are indicated with the following symbols:\n  \x1b[32m+\x1b[0m create\x1b[0m\n\nOpenTofu will perform the following actions:\n\n\x1b[1m  # module.vpc.random_integer.example[0]\x1b[0m will be created\n\x1b[0m  \x1b[32m+\x1b[0m\x1b[0m resource \"random_integer\" \"example\" {\n      \x1b[32m+\x1b[0m\x1b[0m id      = (known after apply)\n      \x1b[32m+\x1b[0m\x1b[0m keepers = {\n          \x1b[32m+\x1b[0m\x1b[0m \"example\" = \"hello world\"\n        }\n      \x1b[32m+\x1b[0m\x1b[0m max     = 50000\n      \x1b[32m+\x1b[0m\x1b[0m min     = 1\n      \x1b[32m+\x1b[0m\x1b[0m result  = (known after apply)\n    }\n\n\x1b[1mPlan:\x1b[0m 1 to add, 0 to change, 0 to destroy.\n\x1b[0m\x1b[90m\n─────────────────────────────────────────────────────────────────────────────\x1b[0m\n\nNote: You didn't use the -out option to save this plan, so OpenTofu can't\nguarantee to take exactly these actions if you run \"tofu apply\" now.\n" should not contain "Initializing modules"
2026-05-08T16:08:27.3327620Z         	Test:       	TestTerragruntInitOnce
2026-05-08T16:08:27.3327780Z         	Messages:   	init command executed more than once
2026-05-08T16:08:27.3327850Z --- FAIL: TestTerragruntInitOnce (4.25s)
```

--- a/docs/src/data/changelog/v1.0.5/auto-init-marker-cached-modules.mdx
+++ b/docs/src/data/changelog/v1.0.5/auto-init-marker-cached-modules.mdx
@@ -1,0 +1,10 @@
+---
+version: "v1.0.5"
+category: "bug-fixes"
+---
+
+#### Auto-init now re-runs after a source change when modules are already cached
+
+`terragrunt plan`/`apply` could fail with `Error: Required plugins are not installed` after a source-version change in any unit with a `module ""` block. The `.terragrunt-init-required` marker written on source change was being ignored because `modulesNeedInit` short-circuited as soon as `.terraform/modules/` existed.
+
+The marker check now lives at the top of `needsInitRunCfg` and is honored regardless of cached `.terraform/modules/` contents.

--- a/test/fixtures/init-once/terragrunt.hcl
+++ b/test/fixtures/init-once/terragrunt.hcl
@@ -1,3 +1,3 @@
 terraform {
-  source = ".//module"
+  source = "./module"
 }

